### PR TITLE
Spamassassin channels parser

### PR DESCRIPTION
**Pull request by skontar**
Specs and parser for spamassassin channels.

### CODE REVIEW

**Comment by skontar** [comment]
https://gitlab.cee.redhat.com/insights-release-eng/insights-core-assets/-/merge_requests/49

File: insights/parsers/spamassassin_channels.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+
+@parser(Specs.spamassassin_channels)
+class SpamassassinChannels(CommandParser):
+    """
+    Class for parsing the ``/bin/grep '^\\s*CHANNELURL=' /etc/mail/spamassassin/channel.d/*.conf``
+    command.
+
+    Attributes:
+        channels (OrderedDict): channels grupped by configuration file
```
[the remark this answers is not in the document]
Comment (skontar) [reply insights/parsers/spamassassin_channels.py:20]: Yes :facepalm: .

File: insights/parsers/spamassassin_channels.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    def parse_content(self, content):
+        channels = OrderedDict()
+        for line in content:
+            file_name, file_line = line.split(":", 1)
+            channel = re.sub('^\\s*CHANNELURL=', '', file_line).strip()
+            if file_name not in channels:
+                channels[file_name] = []
+            channels[file_name].append(channel)
+
+        self.channels = channels
```
Comment (jobselko) [inline insights/parsers/spamassassin_channels.py:44]: What is the meaning of this line and line 33? I am just curious.

File: insights/parsers/tests/test_spamassassin_channels.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+TWO_IN_FILE = """
+/etc/mail/spamassassin/channel.d/sought.conf:CHANNELURL=sought.rules.yerp.org
+/etc/mail/spamassassin/channel.d/sought.conf:CHANNELURL=sought2.rules.yerp.org
+/etc/mail/spamassassin/channel.d/spamassassin-official.conf:CHANNELURL=updates.spamassassin.org
+""".strip()
+
+
+TEST_CASES = [
+    (DEFAULT, {
+        "/etc/mail/spamassassin/channel.d/sought.conf": ["sought.rules.yerp.org"],
```
Comment (jobselko) [inline insights/parsers/tests/test_spamassassin_channels.py:27]: I would use variables for both input and output to have it unified, e.g. (`DEFAULT`, `EXPECTED_DEFAULT`) etc.

File: insights/parsers/spamassassin_channels.py (hunk 2 above)
Comment (skontar) [reply insights/parsers/spamassassin_channels.py:44]: If there is an exception, the `self.channels` will stay None. However using class variable is likely wrong, that should have been done in the init.

File: insights/parsers/tests/test_spamassassin_channels.py (hunk 3 above)
Comment (skontar) [reply insights/parsers/tests/test_spamassassin_channels.py:27]: Good idea.

File: insights/specs/default.py [hunk 4]
```diff
@@ -596,6 +596,7 @@ def sap_sid_name(broker):
     sockstat = simple_file("/proc/net/sockstat")
     softnet_stat = simple_file("proc/net/softnet_stat")
     software_collections_list = simple_command('/usr/bin/scl --list')
+    spamassassin_channels = simple_command("/bin/grep '^\\s*CHANNELURL=' /etc/mail/spamassassin/channel.d/*.conf")
```
Comment (bfahr) [inline insights/specs/default.py:599]: @skontar there is no shell to interpret the `*.conf` file glob so this command won't work.

**Pull request merged**